doc/decisions/1_in_discussion/reference_counting.md

```diff
@@ -0,0 +1,286 @@
+# Reference Counting
+
+## Problem
+
+Elektra's data structures are shared between multiple owners.
+A `Key` can be owned by many different `KeySet`s.
+Both `Key` and `KeySet` can be used in bindings.
+If the languages of those bindings have automatic memory management, this also implies a shared ownership.
+
+To solve this issue of shared ownership, we need reference counting.
```

Comment:
This is AFAIK already solved in the current API. See also https://github.com/ElektraInitiative/libelektra/pull/4805#discussion_r1083989993

Reply:
See https://github.com/ElektraInitiative/libelektra/pull/4676#discussion_r1088275260